doc/api/stream.md

```diff
@@ -1169,6 +1169,30 @@ The `writable.\_writev()` method is prefixed with an underscore because it is
 internal to the class that defines it, and should never be called directly by
 user programs.
 
+#### Errors While Writing
+
+It is recommended that errors occurring during the processing of the
+`writable.\_write()` and `writable.\_writev()` methods are reported by invoking
+the callback and passing the error as the first argument. This will cause an
+`'error'` event to be emitted by the Writable. Throwing an Error from within
+`writable.\_write()` can result in expected and inconsistent behavior depending
+on how the stream is being used.  Using the callback ensures consistent and
+predictable handling of errors.
+
+```js
+const Writable = require('stream').Writable;
+
+const myWritable = new Writable({
+  write(chunk, encoding, callback) {
+    try {
+      // do some work
+    } catch (err) {
+      callback(err);
+    }
+  }
+});
+```
```

Comment:
This example states "you should use try-catch within `_write`", but then newbies can put some async stuff inside the try-catch.  How about avoiding the try-catch?

``````js
 const Writable = require('stream').Writable;
 
 const myWritable = new Writable({
   write(chunk, encoding, callback) {
     if (chunk.toString().indexOf('a') >= 0) {
       callback(new Error('chunk is invalid'))
     } else {
        callback()
     }
   }
 });
```